src/filter.rs

```diff
@@ -134,6 +136,65 @@ impl<D: Ord> KmerSummarizer<D, SmallVec<[D; 4]>> for CountFilterSmallInt<D> {
     }
 }
 
+//Equivalence class bases implementation
+pub type EqClassIdType = u32 ;
+pub struct CountFilterEqClass<D: Eq + Hash> {
+    min_kmer_obs: usize,
+    eq_classes: HashMap<Vec<D>, EqClassIdType>,
+}
+
+impl<D: Eq + Hash> CountFilterEqClass<D> {
+    pub fn new(min_kmer_obs: usize) -> CountFilterEqClass<D> {
+        CountFilterEqClass {
+            min_kmer_obs: min_kmer_obs,
+            eq_classes: HashMap::<Vec<D>, EqClassIdType>::new(),
+        }
+    }
+    pub fn get_eq_classes(self) -> HashMap<Vec<D>, EqClassIdType>{
+        self.eq_classes
+    }
+}
+
+/// Implement this trait to control how multiple observations of a kmer
+/// are carried forward into a DeBruijn graph.
+pub trait MutKmerSummarizer<DI, DO> {
+    /// The input `items` is an iterator over kmer observations. Input observation
+    /// is a tuple of (kmer, extensions, data). The summarize function inspects the
+    /// data and returns a tuple indicating:
+    /// * whether this kmer passes the filtering criteria (e.g. is there a sufficient number of observation)
+    /// * the accumulated Exts of the kmer
+    /// * a summary data object of type `DO` that will be used as a color annotation in the DeBruijn graph.
+    fn summarize<K, F: Iterator<Item = (K, Exts, DI)>>(&mut self, items: F) -> (bool, Exts, DO);
```

Comment:
nice!  let's just change the original KmerSummarizer trait to be &mut self, rather than introduce a new trait.

Reply:
Sounds good, thanks for taking a quick look !!